site/source/docs/api_reference/vr.h.rst

```diff
@@ -0,0 +1,378 @@
+.. _vr-h:
+
+====
+vr.h
+====
+
+The C APIs in `vr.h <https://github.com/kripken/emscripten/blob/master/system/include/emscripten/vr.h>`_ provide basic interfaces for interacting with WebVR from Emscripten.
+
+.. contents:: Table of Contents
+	:local:
+	:depth: 1
+
+.. _test-example-code-vr-api:
+
+Test/Example code
+-----------------
+
+The vr test code demonstrates how to use this API:
+
+	- `test_vr.c <https://github.com/kripken/emscripten/blob/master/tests/test_vr.c>`_
+
+.. _functions-vr-api:
+
+Functions
+---------
+
+Initialization
+==============
+
+.. c:function:: int emscripten_vr_init()
+
+	Initialize the emscripten VR API. This will `navigator.getVRDisplays()
+	<https://w3c.github.io/webvr/spec/1.1/#navigator-getvrdisplays-attribute>`_
+	and when completed, set the return of :c:func:`emscripten_vr_ready` to be `true`.
+
+	:returns: `1` on success, `0` if the browsers WebVR support is insufficient.
+	:rtype: int
+
+.. tip:: This call succeeding is not sufficient for use of the rest of the API. Please
+	make sure to wait until the return value of :c:func:`emscripten_vr_ready` is true.
+
+.. c:function:: int emscripten_vr_ready()
+
+	Check whether the VR API has finished initializing VR displays.
```

Comment:
Yikes, polling "is init done?" API - we'd want a waitable handle here, which a token that one can likewise poll, wait or get a "when it's done" callback for. Though full support for that will need multithreading enabled.

Reply:
Yeah, I didn't really like this part either...

> Though full support for that will need multithreading enabled.

You are currently working on this, right? For now I could just add a callback to the `emscripten_vr_init()` function.

Reply:
Yeah, that's right - that's why it's fresh on my mind right now, though it's not quite landed just yet to be usable, so it will be a moment until that gets in to be used by other APIs like this, but I'll note when that is.

Reply:
I added a callback for now. Should I add a TODO comment or similar for making this a waitable handle?